core/string/ustring.cpp

```diff
@@ -3107,6 +3107,116 @@ String String::erase(int p_pos, int p_chars) const {
 	return left(p_pos) + substr(p_pos + p_chars);
 }
 
+String String::remove_char(char32_t p_char) const {
+	int len = length();
+
+	if (p_char == 0 || len == 0) {
+		return *this;
+	}
+
+	int index = find_char(p_char);
+
+	// If no occurrence of `char` was found, return this.
+	if (index == -1) {
+		return *this;
+	}
+
+	const char32_t *old_ptr = ptr();
+
+	// If we found at least one occurrence of `char`, create new string, allocating enough space for the current length minus one.
+	String new_string;
+	new_string.resize(len);
+	char32_t *new_ptr = new_string.ptrw();
+
+	// Copy part of input before `char`.
+	memcpy(new_ptr, old_ptr, index * sizeof(char32_t));
+
+	int new_size = index;
+
+	// Copy rest, skipping `char`.
+	for (++index; index < len; ++index) {
+		const char32_t old_char = old_ptr[index];
+		if (old_char != p_char) {
+			new_ptr[new_size] = old_char;
+			++new_size;
+		}
+	}
+
+	new_ptr[new_size] = _null;
+
+	// Shrink new string to fit.
+	new_string.resize(new_size + 1);
+
+	return new_string;
+}
+
+template <class T>
+static bool _contains_char(char32_t p_c, const T *p_chars, int p_chars_len) {
```

Comment:
This deserves to be a separate PR as multiple-char `find` would definitely be useful elsewhere too. Though I wouldn't mind having this function ~ temporarily ~ to account for it not being here yet :)